apps/dotcom/sync-worker/src/TLUserDurableObject.ts

```diff
@@ -389,11 +389,15 @@ export class TLUserDurableObject extends DurableObject<Environment> {
 			isApp: true,
 		})
 		const publishedHistory = await listAllObjectKeys(this.env.ROOM_SNAPSHOTS, publishedPrefixKey)
-		await this.env.ROOM_SNAPSHOTS.delete(publishedHistory)
+		if (publishedHistory.length > 0) {
+			await this.env.ROOM_SNAPSHOTS.delete(publishedHistory)
+		}
 		// remove edit history
 		const r2Key = getR2KeyForRoom({ slug: id, isApp: true })
 		const editHistory = await listAllObjectKeys(this.env.ROOMS_HISTORY_EPHEMERAL, r2Key)
-		await this.env.ROOMS_HISTORY_EPHEMERAL.delete(editHistory)
+		if (editHistory.length > 0) {
+			await this.env.ROOMS_HISTORY_EPHEMERAL.delete(editHistory)
+		}
```

Comment:
this fixes the 'file not deleting' bug. in remote envs R2 complains if you call .delete with an empty array. i saw the errors popping up in sentry.

what's odd is that the mutations didn't seem to be rejected. I think that may not be working remotely too. i.e. if an error is thrown during handleMutation, inside the transaction maybe, it seems to crap out.

Reply:
added ticket here https://linear.app/tldraw/issue/INT-520/investigate-mutation-rejection